pulp_service/pulp_service/app/viewsets.py

```diff
@@ -148,19 +151,33 @@ def view_name(cls):
 
 class VulnerabilityReport(ViewSet):
 
-    def list(self, request):
-        queryset = VulnReport.objects.all()
+    def retrieve(self, request):
+        queryset = VulnReport.objects.filter(pulp_domain=get_domain())
         serializer = VulnerabilityReportSerializer(queryset, many=True)
         return Response(serializer.data)
 
-    def get(self, request, uuid):
-        queryset = VulnReport.objects.get(id=uuid)
+    def list(self, request, uuid):
+        try:
+            queryset = VulnReport.objects.get(id=uuid, pulp_domain=get_domain())
```

Comment:
`id` is unique regardless of domain so you don't need to specify `pulp_domain` for this `get`.
```suggestion
            queryset = VulnReport.objects.get(id=uuid)
```